src/net_processing.cpp

```diff
@@ -4335,20 +4337,20 @@ void PeerManagerImpl::CheckForStaleTipAndEvictPeers()
 {
     LOCK(cs_main);
 
-    int64_t time_in_seconds = GetTime();
+    auto now{GetTime<std::chrono::seconds>()};
 
-    EvictExtraOutboundPeers(std::chrono::seconds{time_in_seconds});
+    EvictExtraOutboundPeers(now);
 
-    if (time_in_seconds > m_stale_tip_check_time) {
+    if (now > m_stale_tip_check_time) {
         // Check whether our tip is stale, and if so, allow using an extra
         // outbound peer
         if (!fImporting && !fReindex && m_connman.GetNetworkActive() && m_connman.GetUseAddrmanOutgoing() && TipMayBeStale()) {
-            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", time_in_seconds - m_last_tip_update);
+            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", count_seconds(now) - count_seconds(m_last_tip_update));
```

Comment:
nit: I think it might have been better to do the calculation type-safe and then only "serialize" the result of the addition. This would also result in less code.

Reply:
One doubt:

Is there a way to go from `atomic<seconds>` to `<seconds>`
Or do subtraction between two `atomic<seconds>` type variables?

Thanks.

Reply:
>  subtraction between two atomic<seconds> type variables?

For reference, doing so would cause the following error:
```
no match for ‘operator-’ (operand types are ‘std::atomic<std::chrono::duration<long int> >’ and ‘std::atomic<std::chrono::duration<long int> >’)
```

cc @MarcoFalke